### ISSUE TITLE
Sass compilation issues cause 'Cannot read property postcss of null'

### ISSUE THREAD

**Issue by jameshulse** [mzgoddard/hard-source-webpack-plugin#238]
If I get a sass compilation error then even after fixing the problem I will get the below issue on all subsequent builds. Deleting `node_modules/.cache` fixes the issue.

```
in ./resources/assets/sass/app.scss

Module build failed: ModuleBuildError: Module build failed: TypeError: Cannot read property 'postcss' of null
    at Processor.normalize (.../node_modules/postcss-loader/node_modules/postcss/lib/processor.js:136:13)
    at new Processor (.../node_modules/postcss-loader/node_modules/postcss/lib/processor.js:51:25)
    at postcss (.../node_modules/postcss-loader/node_modules/postcss/lib/postcss.js:73:10)
    at Promise.resolve.then.then (.../node_modules/postcss-loader/lib/index.js:146:12)
    at <anonymous>
    at runLoaders (.../node_modules/webpack/lib/NormalModule.js:195:19)
    at .../node_modules/loader-runner/lib/LoaderRunner.js:364:11
    at .../node_modules/loader-runner/lib/LoaderRunner.js:230:18
    at context.callback (.../node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at Promise.resolve.then.then.catch (.../node_modules/postcss-loader/lib/index.js:189:71)
    at <anonymous>
```

I am using laravel-mix (a wrapper for webpack) which doesn't allow me to easily show you my full webpack config - I've posted on their github to find out if I can get a dump of the config to post here.

**Comment by stevefan1999** [comment]
I'm also having this issue while I don't even have any LESS/SASS transforms at all...

**Comment by mzgoddard** [comment]
Thanks for filing this issue @jameshulse, and your side note @stevefan1999. @jameshulse that stack trace is super helpful. I think there is another issue for this but I wasn't able to get anyone to report a stack trace for it.

**Comment by mzgoddard** [comment]
Ah no that was a separate development while working on an not directly related bug.

**Comment by jameshulse** [comment]
No problem @mzgoddard. Do you want me to try and extract my webpack config or is this enough to go on for now?

**Comment by ypconstante** [comment]
I used to have this problem too, the solution was to add the `ident` option in the `postcss-loader` config (https://github.com/postcss/postcss-loader#plugins).

**Comment by filip-savic** [comment]
@mzgoddard cc @jameshulse @stevefan1999 

I'm getting a different error, but I believe I had the one @jameshulse described, but I can't reproduce it now.

Also, adding the `ident` option did not work for me.

The error I'm getting is:

```
ERROR in ./node_modules/postcss-loader/lib?{}!./style.src.css
Module build failed: Error: No PostCSS Config found in: C:\hardsource-issue-238
    at C:\hardsource-issue-238\node_modules\postcss-load-config\index.js:51:26
    at <anonymous>
 @ ./style.src.css 4:14-95
```

-----

I've set up a repo only with the essentials so you can easily reproduce it - https://github.com/filip-savic/hardsource-issue-238/commits/master (just revert the "Workaround" commit).

Steps to reproduce error:

1. run initial build (`npm run build` in the repo)
2. change the css file
3. run build again 

-----

The problem seems to be in `postcss-loader`:

- https://github.com/postcss/postcss-loader/issues/281
- https://github.com/postcss/postcss-loader/issues/204

The workaround for me was to add a `postcss.config.js` file instead of the inline options in `webpack.config.js`.

**Comment by ppoulsen** [comment]
I believe this was fixed in Webpack v3.11.0 by webpack/webpack#6173! 🎉 Previously, this was getting written to the cache on the first run:
```
{
  "context":"/path/to/hardsource-issue-238",
  "request":"/path/to/hardsource-issue-238/node_modules/postcss-loader/lib/index.js?{}!/path/to/hardsource-issue-238/style.src.css",
  "dependencies":null,
  "userRequest":"/path/to/hardsource-issue-238/node_modules/postcss-loader/lib/index.js?{}!/path/to/hardsource-issue-238/style.src.css",
  "rawRequest":"!!./node_modules/postcss-loader/lib/index.js??ref--0-1!./style.src.css",
  "loaders":[
    {
      "loader":"/path/to/hardsource-issue-238/node_modules/postcss-loader/lib/index.js",
      "options":{

      }
    }
  ],
  "resource":"/path/to/hardsource-issue-238/style.src.css",
  "parser":null,
  "parserOptions":{

  }
}
```
Notice no `ident` for the loader, which would fail to look up the options [here](https://github.com/mzgoddard/hard-source-webpack-plugin/blob/d58eca92a35e81ab5a5b52aa745cca80468408bd/index.js#L1189-L1196) on the second run, causing postcss-loader to try to find a postcss.config.js file.

**Comment by jameshulse** [comment]
Can confirm that it was fixed for me after upgrading to webpack 3.11.0. Thanks @ppoulsen.
